diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/identifiers/LocalVariablesRule.kt

```diff
@@ -112,15 +113,36 @@ class LocalVariablesRule(private val configRules: List<RulesConfig>) : Rule("loc
         properties
             .sortedBy { it.node.getLineNumber() }
             .zip(
-                (properties.size - 1 downTo 0).map { it + numLinesAfterLastProp }
-            )
-            .forEachIndexed { index, (property, offset) ->
-                if (index != properties.lastIndex) {
-                    checkLineNumbers(property, statement.node.getLineNumber(), offset)
-                } else {
-                    // since offset after last property is calculated in this method, we pass offset = 0
-                    checkLineNumbers(property, statement.node.getLineNumber(), 0)
+                (properties.size - 1 downTo 0).map { index ->
+                    val siblings = properties
+                        .sortedBy { it.node.getLineNumber() }[properties.lastIndex - index]
+                        .siblings(forward = true, withItself = false)
+
+                    // Also we need to count number of comments to skip. See `should skip comments` test
+                    // For the last property we don't need to count, because they will be counted in checkLineNumbers
+                    // We count number of comments beginning from next property
+                    val numberOfComments = siblings
+                        .takeWhile { it != statement }
+                        .dropWhile { it !is KtProperty }
+                        .filter { it.node.isPartOfComment() }
+                        .count()
+
+                    // We should also skip all vars that were not included in properties list, but they are between statement and current property
+                    val numberOfVarWithInitializer = siblings
+                        .takeWhile { it != statement }
+                        .filter { it is KtProperty && it.isVar && it.initializer != null && it !in properties }
```

Comment:
```suggestion
                        .filter { it is KtProperty && it !in properties }
```
Maybe, it's enough to count properties that are not included in our list? We may change the inclusion criterion in the future, but here we still will need to count all other properties. Also, I don't see why `val`s shouldn't be included here.